snpdb/models/models_variant.py

```diff
@@ -242,32 +242,109 @@ class AlleleMergeLog(TimeStampedModel):
     message = models.TextField(null=True)
 
 
-@dataclass(frozen=True)
```

Comment:
Removing frozen=True removes the hash implementation (otherwise you have to say @dataclass(unsafe_hash=True)

Is it only no longer frozen so you could do validation in the constructor?
If so, that's better done with `def __post_init__(self):`

but I also see that end optional and can be calculated with ref and alt (in some cases), I'd recommend a classmethod factory method there, though there is a LOT of references to VariantCoordiante( ) I think it's worth it to make VariantCoordinate immutable

Reply:
Yeah the very few times I need to do it without and end can be via classmethod